include/drivers/pcie/pcie.h

```diff
@@ -193,8 +192,16 @@ extern void pcie_irq_enable(pcie_bdf_t bdf, unsigned int irq);
 #define PCIE_CONF_BAR_MEM(w)		(((w) & 0x00000001U) != 0x00000001U)
 #define PCIE_CONF_BAR_64(w)		(((w) & 0x00000006U) == 0x00000004U)
 #define PCIE_CONF_BAR_ADDR(w)		((w) & ~0xfUL)
+#define PCIE_CONF_BAR_FLAGS(w)		((w) & 0xfUL)
 #define PCIE_CONF_BAR_NONE		0U
 
+#define PCIE_CONF_BAR_INVAL		0xFFFFFFF0U
+#define PCIE_CONF_BAR_INVAL64		0xFFFFFFFFFFFFFFF0U
+
+#define PCIE_CONF_BAR_INVAL_FLAGS(w)			\
+	((((w) & 0x00000006U) == 0x00000006U) ||	\
+	 (((w) & 0x00000006U) == 0x00000002U))
```

Comment:
is size <1Mb considered invalid aswell?

Reply:
yes, per PCIE specs this value is now reserved and does no longer mean anything.